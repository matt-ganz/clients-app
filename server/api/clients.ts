import * as clients from "./clients.json";

export default defineEventHandler(() => {
  return {
    clients,
  };
});
