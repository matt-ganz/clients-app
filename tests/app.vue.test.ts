import { describe, it, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import App from "../app.vue";
import * as clients from "../server/api/clients.json";

describe("import app component succeeds", () => {
  test("normal imports as expected", async () => {
    const cmp = await import("../app.vue");
    expect(cmp).toBeDefined();
  });
});

describe("app component initializes successfully", () => {
  test("app initializes as expected", async () => {
    expect(App).toBeTruthy();
  });
});

describe("app component mounts successfully and matches snapshot", () => {
  test("app initializes as expected", async () => {
    const wrapper = mount(App);
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe("search clients", () => {
  test("should return default value of search term (and all clients)", async () => {
    const wrapper = mount(App);
    const myClients = clients;
    wrapper.vm.filterText = "";

    expect(wrapper.vm.filterText).toBe("");
    expect(myClients).toEqual(clients);
  });
});
