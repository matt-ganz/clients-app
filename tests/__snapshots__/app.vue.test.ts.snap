// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app component initializes successfully > app initializes as expected 1`] = `""`;

exports[`app component mounts successfully and matches snapshot > app initializes as expected 1`] = `""`;
